src/main/java/nwoolcan/controller/ProductionControllerImpl.java

```diff
@@ -0,0 +1,34 @@
+package nwoolcan.controller;
+
+import nwoolcan.model.brewery.Brewery;
+import nwoolcan.model.brewery.production.batch.QueryBatch;
+import nwoolcan.model.brewery.production.batch.QueryBatchBuilder;
+import nwoolcan.viewmodel.brewery.production.ProductionViewModel;
+import nwoolcan.viewmodel.brewery.production.batch.MasterBatchViewModel;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+//Package-private.
+final class ProductionControllerImpl implements ProductionController {
+
+    private final Brewery brewery;
+
+    //Package-private.
+    ProductionControllerImpl(final Brewery brewery) {
+        this.brewery = brewery;
+    }
+
+    @Override
+    public ProductionViewModel getViewModel() {
+        return new ProductionViewModel(this.brewery.getBatches(new QueryBatchBuilder().build().getValue()));
+    }
+
+    @Override
+    public List<MasterBatchViewModel> getBatches(final QueryBatch query) {
```

Comment:
I think this should be a method of the brewery, not of the productionController (if we want to follow the structure of the model...)

Reply:
Actually the productionController should not exist if we want to follow the structure of the model.. otherwise it may is good to say that to discuss it with who wrote the brewery...